docs/DesignDocs/NotesOnAttestationAPI.md

```diff
@@ -0,0 +1,366 @@
+is designed to be TEE-agnostic.
+
+- The OE SDK is designed for attester and verifier software to be TEE agnostic,
+so that the same source code can be built for and run on different platforms
+- But owners of attesters and verifiers define their security appraisal policies,
+and the policies are not TEE agnostic. For example:
+  - The policies define which TEEs, platforms and configurations are trusted
+  to run the attesters and verifiers.
+    - They also define security properties of these entities, e.g. keypairs
+    and certs, service provider ID / SPID for SGX EPID evidence, etc.
+    These security properties are required for attestation and secure
+    communication.
+  - The policies define the list of evidence formats acceptable to the verifiers.
+    - Many acceptable evidence format are TEE specific. Examples include
+    SGX location attestation evidence formats, SGX ECDSA evidence formats,
+    and SGX EPID evidence formats.
+- The OE SDK provides means for the attesters and verifiers to be built and
+configured for specified TEEs and platforms, and to generate or verify evidences
+in the list of acceptable formats as defined in the appraisal policies.
+  - The OE SDK attestation API enables the attesters and verifiers to register
+  the plugins that are built, configured, and working
+  properly, and enumerate the supported list of evidence formats covered by
+  these plugins.
+  For a specific attester or verifier instance, the supported list
+  of evidence formats is not required to perfectly match the acceptable list
+  defined in the appraisal policies.
+
+# Notes on SGX Plugins Design
+
+## Evidence Format, Plugin, and Plugin Library
+
+This section explains various terms used in the OE SDK attestation API
+definition and plugin-based design for OE SDK implementation of these APIs:
+* Evidence format, as exposed by the [OE SDK attestation API](https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/Attestation_API_Proposal.md).
+* Plugin, as a logical unit with a [well-defined set of entry points](https://github.com/openenclave/openenclave/blob/master/include/openenclave/internal/plugin.h),
+to support generation or verification of evidence identified by an evidence format.
+* Plugin library, for implementation of one or more plugins.
+
+In the proposed [OE SDK attestation API](https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/Attestation_API_Proposal.md),
+every evidence format is uniquely identified by a UUID. Enclave and host
+applications can discover formats supported for evidence verification.
+Enclave applications can select a format for evidence generation. For evidence
+generation, API `oe_get_evidence()` takes in an evidence format UUID along with
+other input parameters. Upon successful operation, `oe_get_evidence()` returns
+an evidence blob with the evidence format UUID included as a field in its
+header structure.
```

Comment:
This sentence is about to become false with PR #3198, so reword.
```suggestion
a blob of evidence blob in the indicated format.  If the evidence is wrapped with an OE evidence header, the evidence format UUID included as a field in the
header structure.
```

Reply:
Thanks!